encryption-client/src/main/java/org/wso2/mi/tooling/security/utils/IOUtils.java

```diff
@@ -0,0 +1,126 @@
+/*
+ * Copyright (c) 2020, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ * WSO2 Inc. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.wso2.mi.tooling.security.utils;
+
+import org.wso2.mi.tooling.security.exception.EncryptionToolException;
+import org.wso2.mi.tooling.security.output.K8SecretYaml;
+import org.yaml.snakeyaml.DumperOptions;
+import org.yaml.snakeyaml.Yaml;
+
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.FileOutputStream;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.nio.file.Paths;
+import java.util.Properties;
+
+public class IOUtils {
+
+    /**
+     * Get properties defined in file.
+     *
+     * @param fileName properties file name
+     * @return Properties object
+     */
+    public static Properties getProperties(String fileName) {
+        Properties properties =  new Properties();
+        File propertiesFile = new File(fileName);
+        if (!propertiesFile.exists()) {
+            throw new EncryptionToolException(fileName + " file cannot be found");
+        }
+
+        InputStream in = null;
+        try {
+            in = new FileInputStream(propertiesFile);
+            properties.load(in);
+        } catch (IOException error) {
+            throw new EncryptionToolException("Error loading properties from file");
+        } finally {
+            if (in != null) {
+                try {
+                    in.close();
+                } catch (IOException ignored) {
+                    throw new EncryptionToolException("Error while closing input stream");
```

Comment:
The input parameter says it's ignored, but we throw an EncryptionToolException. Is this intentional?

Reply:
This is fixed with try-with resources.